### PR TITLE
Document component descriptions/props

### PR DESCRIPTION
**Pull request by andydavison**
- Adds short descriptions for each component/provider
- Adds argTypes descriptions for components/providers
- Adds separate docs for `TableColumn`

### CODE REVIEW

**Comment by ajacksified** [comment]
Looks good - just missing prop descriptions on LogoutButton.

**Comment by andydavison** [comment]
> Looks good - just missing prop descriptions on LogoutButton.

@ajacksified just fixed — apparently the description doesn't show unless I also set `control: { type: null }` 🤷‍♂️

**Comment by andydavison** [comment]
I've also added TableColumn docs in — annoyingly there doesn't seem to be a way to use the 'subcomponent' documentation _and_ have customised argTypes for the subcomponent, so it's a separate story to `Table`, with a note saying go look at `Table` for the full examples.

File: stories/components/link.stories.tsx [hunk 1]
```diff
@@ -26,6 +26,15 @@ import { Link } from "../../src/components/link";
 export default {
   title: "Components/Link",
   component: Link,
+  argTypes: {
+    property: {
+      type: { required: true },
+      description: `The property of the Thing to retrieve the link URL from.`,
```
Comment (nicolasmondada) [inline stories/components/link.stories.tsx:32]:
```suggestion
      description: `The property of the [Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing) to retrieve the link URL from.`,
```

File: src/components/link/index.tsx [hunk 2]
```diff
@@ -27,6 +27,9 @@ export type Props = {
   property: Url | UrlString;
 } & React.AnchorHTMLAttributes<HTMLAnchorElement>;
 
+/**
+ * Retrieves a URL from given Thing/property, and renders as an anchor tag with the given href.
```
Comment (nicolasmondada) [inline src/components/link/index.tsx:31]:
```suggestion
 * Retrieves a URL from given [Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing)/property, and renders as an anchor tag with the given href.
```

File: stories/components/image.stories.tsx [hunk 3]
```diff
@@ -30,6 +30,45 @@ const { host } = config();
...
   title: "Components/Image",
   component: Image,
+  argTypes: {
+    thing: {
+      description: `The [Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing) to retrieve the image src from. Uses a Thing from context if not supplied.`,
+      control: { type: null },
+    },
+    property: {
+      type: { required: true },
+      description: `The property of the Thing to retrieve the src URL from.`,
```
Comment (nicolasmondada) [inline stories/components/image.stories.tsx:40]:
```suggestion
      description: `The property of the [Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing) to retrieve the src URL from.`,
```

File: stories/components/text.stories.tsx [hunk 4]
```diff
@@ -32,8 +32,39 @@ export default {
...
+    },
+    thing: {
+      description: `The [Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing) to retrieve the text from. Uses a Thing from context if not supplied.`,
+    },
+    dataSet: {
+      description: `The [Dataset](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-SolidDataset) to retrieve the text from. Uses a Dataset from context if not supplied.`,
+    },
+    property: {
+      type: { required: true },
+      description: `The property of the Thing to retrieve the text from.`,
```
Comment (nicolasmondada) [inline stories/components/text.stories.tsx:51]:
```suggestion
      description: `The property of the [Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing) to retrieve the text from.`,
```

File: stories/components/value.stories.tsx [hunk 5]
```diff
@@ -57,6 +65,31 @@ export default {
...
     },
+    thing: {
+      description: `The [Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing) to retrieve the value from. Uses a Thing from context if not supplied.`,
+    },
+    dataSet: {
+      description: `The [Dataset](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-SolidDataset) to retrieve the value from. Uses a Dataset from context if not supplied.`,
+    },
+    property: {
+      type: { required: true },
+      description: `The property of the Thing to retrieve the value from.`,
```
Comment (nicolasmondada) [inline stories/components/value.stories.tsx:76]:
```suggestion
      description: `The property of the [Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing) to retrieve the value from.`,
```

File: stories/components/video.stories.tsx [hunk 6]
```diff
@@ -26,6 +26,45 @@ import { Video } from "../../src/components/video";
...
   title: "Components/Video",
   component: Video,
+  argTypes: {
+    thing: {
+      description: `The [Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing) to retrieve the video src from. Uses a Thing from context if not supplied.`,
+      control: { type: null },
+    },
+    property: {
+      type: { required: true },
+      description: `The property of the Thing to retrieve the src URL from.`,
```
Comment (nicolasmondada) [inline stories/components/video.stories.tsx:36]:
```suggestion
      description: `The property of the [Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing) to retrieve the src URL from.`,
```

File: src/components/image/index.tsx [hunk 7]
```diff
@@ -37,6 +37,9 @@ export type Props = {
   errorComponent?: React.ComponentType<{ error: Error }>;
 } & React.ImgHTMLAttributes<HTMLImageElement>;
 
+/**
+ * Fetches and displays an image, from a src found at a given property of a given Thing. Can also be used to upload a new/replacement image file.
```
Comment (nicolasmondada) [inline src/components/image/index.tsx:41]:
```suggestion
 * Fetches and displays an image, from a src found at a given property of a given [Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing). Can also be used to upload a new/replacement image file.
```

File: src/components/table/index.tsx [hunk 8]
```diff
@@ -47,6 +47,10 @@ export type TableColumnProps = {
   filterable?: boolean;
   children?: undefined | null | [];
 };
+
+/**
+ * To be used as the only children of a Table component. Each column represents one property of the Things passed to the Table.
```
Comment (nicolasmondada) [inline src/components/table/index.tsx:52]:
```suggestion
 * To be used as the only children of a [`Table`](/?path=/docs/components-table)

 component. Each column represents one property of the [Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing) passed to the [`Table`](/?path=/docs/components-table).
```

File: src/components/table/index.tsx [hunk 9]
```diff
@@ -68,6 +72,9 @@ export interface TableProps
   ) => React.HTMLAttributes<HTMLTableRowElement>;
 }
 
+/**
+ * Displays values from an array of Things as table rows, with each column showing a given property of those Things.
```
Comment (nicolasmondada) [inline src/components/table/index.tsx:76]:
```suggestion
 * Displays values from an array of [Things](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing) as table rows, with each column showing a given property of those [Things](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing).
```

File: src/components/text/index.tsx [hunk 10]
```diff
@@ -51,6 +51,9 @@ export type Props = {
   onError?(error: Error): void | null;
 };
 
+/**
+ * Retrieves and displays a string from a given Dataset/Thing/property. Can also be used to set/update and persist a string value.
```
Comment (nicolasmondada) [inline src/components/text/index.tsx:55]:
```suggestion
 * Retrieves and displays a string from a given [Dataset](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-SolidDataset)/[Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing)/property. Can also be used to set/update and persist a string value.
```

File: stories/components/tableColumn.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+    property: {
+      description: `The property of the Things to retrieve the value from and display in the column.`,
+      control: { type: null },
+    },
+    dataType: {
+      description: `The type of value to be retrieved. Uses \`string\` if not set.`,
+      control: { type: null },
+    },
+    header: {
+      description: `A string, or function which returns valid JSX, to render as the column header. Equivalent to the [Header](https://react-table.tanstack.com/docs/api/useTable#column-options) option of React Table.`,
```
Comment (nicolasmondada) [inline stories/components/tableColumn.stories.tsx:50]:
```suggestion
      description: `A string, or function which returns valid JSX, to render as the column header. Equivalent to the [Header](https://react-table.tanstack.com/docs/api/useTable#column-options) option of [React Table](https://www.npmjs.com/package/react-table).`,
```

File: stories/components/tableColumn.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+      control: { type: null },
+    },
+    body: {
+      description: `Function which is passed the [cell props](https://react-table.tanstack.com/docs/api/useTable#cell-properties) and returns valid JSX. Equivalent to the [Cell](https://react-table.tanstack.com/docs/api/useTable#column-options) option of React Table.`,
```
Comment (nicolasmondada) [inline stories/components/tableColumn.stories.tsx:54]:
```suggestion
      description: `Function which is passed the [cell props](https://react-table.tanstack.com/docs/api/useTable#cell-properties) and returns valid JSX. Equivalent to the [Cell](https://react-table.tanstack.com/docs/api/useTable#column-options) option of [React Table](https://www.npmjs.com/package/react-table).`,
```

File: src/components/value/index.tsx [hunk 13]
```diff
@@ -56,6 +56,9 @@ export type Props = {
   onError?(error: Error): void | null;
 } & React.HTMLAttributes<HTMLSpanElement>;
 
+/**
+ * Retrieves and displays a value of one of a range of types from a given Dataset/Thing/property. Can also be used to set/update and persist a value.
```
Comment (nicolasmondada) [inline src/components/value/index.tsx:60]:
```suggestion
 * Retrieves and displays a value of one of a range of types from a given [Dataset](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-SolidDataset)/[Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing)/property. Can also be used to set/update and persist a value.
```

File: src/context/combinedDataContext/index.tsx [hunk 14]
```diff
@@ -65,6 +65,9 @@ function CombinedDataProvider(
   props: DatasetUrl & RequireThingOrThingUrl
 ): ReactElement;
 
+/**
+ * Used to provide both a Dataset and Thing to child components through context, as used by various provided components and the useDataset and useThing hooks.
```
Comment (nicolasmondada) [inline src/context/combinedDataContext/index.tsx:69]:
```suggestion
 * Used to provide both a [Dataset](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-SolidDataset) and [Thing](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-Thing) to child components through context, as used by various provided components and the [`useDataset`](/?path=/docs/hooks-usething) and [`useThing`](/?path=/docs/hooks-usedataset) hooks.
```

File: src/context/datasetContext/index.tsx [hunk 15]
```diff
@@ -55,6 +55,9 @@ export type RequireDatasetOrDatasetUrl =
   | RequireProperty<IDatasetProvider, "dataset">
   | RequireProperty<IDatasetProvider, "datasetUrl">;
 
+/**
+ * Used to provide a Dataset to child components through context, as used by various provided components and the useDataset hook.
```
Comment (nicolasmondada) [inline src/context/datasetContext/index.tsx:59]:
```suggestion
 * Used to provide a [Dataset](https://docs.inrupt.com/developer-tools/javascript/client-libraries/reference/glossary/#term-SolidDataset) to child components through context, as used by various provided components and the useDataset hook.
```

File: src/context/thingContext/index.tsx [hunk 16]
```diff
@@ -54,6 +54,9 @@ export type RequireThingOrThingUrl =
   | RequireProperty<IThingProvider, "thing">
   | RequireProperty<IThingProvider, "thingUrl">;
 
+/**
+ * Used to provide a Thing to child components through context, as used by various provided components and the useThing hook.
```
Comment (nicolasmondada) [inline src/context/thingContext/index.tsx:58]:
```suggestion
 * Used to provide a Thing to child components through context, as used by various provided components and the [`useThing`](/?path=/docs/hooks-usething) hook.
```

**Review by nicolasmondada — CHANGES_REQUESTED** [review]

**Pull request merged**